packages/app/src/app/components/Preview/DevTools/React-Devtools/index.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { initialize } from 'react-devtools-inline/frontend';
+
+import { Container } from './elements';
+
+const DevTools = () => {
+  const ReactDevTools = useRef(null);
+  const [devToolsReady, setDtReady] = useState(false);
+
+  useEffect(() => {
+    const iframe = document.getElementById(
+      'sandbox-preview'
+    ) as HTMLIFrameElement;
+    const contentWindow = iframe.contentWindow;
+    ReactDevTools.current = initialize(contentWindow);
+    iframe.onload = () => {
+      contentWindow.postMessage(
+        {
+          type: 'activate',
+        },
+        '*'
+      );
+    };
+    setDtReady(true);
```

Comment:
You could just store the `DevTools` component directly in state and you'd only need one hook 😁 

```js
const [DevTools, setDevTools] = useState(null);

// in the effect
setDevTools(initialize(contentWindow));
```

Reply:
You can remove the `devToolsReady` state entirely now 😁 If `ReactDevTools` isn't `null`, you know you're ready to roll.

Reply:
YES! All done and updated!

Thanks!